lib/seadsa/DsaLocal.cc

```diff
@@ -543,9 +550,12 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
   IntraBlockBuilder(Function &func, seadsa::Graph &graph, const DataLayout &dl,
                     const TargetLibraryInfo &tli,
                     const seadsa::AllocWrapInfo &allocInfo,
-                    bool track_callsites)
+                    bool track_callsites, bool delay_mem_transfer)
       : BlockBuilderBase(func, graph, dl, tli, allocInfo),
-        m_track_callsites(track_callsites) {}
+        m_track_callsites(track_callsites),
+        m_delay_mem_transfer(delay_mem_transfer) {}
+
+  void proceedMemTransferInsts() { m_delay_mem_transfer = false; }
```

Comment:
Confused about the name. Maybe make it clearer in the name that it sets whether mem transfers are handled eagerly or not